---
layout: distill
title: Introduction to Graphical Models
description: an example of a distill-style lecture notes that showcases the main elements
date: 2019-01-14

lecturers:
  - name: Eric Xing
    url: "https://www.cs.cmu.edu/~epxing/"

authors:
  - name: Daniel Martin  # author's full name
    url: "#"  # optional URL to the author's homepage
  - name: Yiwen Yuan
    url: "#"
  - name: Youngseog Chung
    url: "#"
  - name: Siddharth Satpathy
    url: "#"

editors:
  - name: Maruan Al-Shedivat  # editor's full name
    url: "https://www.cs.cmu.edu/~mshediva/"  # optional URL to the editor's homepage

abstract: >
  In this first lecture, we give an introduction to graphical models. 
We begin by discussing the three fundamental question types that graphical models can help us answer: representation, inference, and learning. 
We then show how graphical models can incorporate domain knowledge, fuse heterogeneous data, and support Bayesian inferences. 
Subsequently, we define the two types of graphical models (directed and undirected) and show that model graph traversal can yield independence relationships (the Equivalence Theorem). 
We conclude with examples of fancier models as well as application areas.
---

## Equations

This theme supports rendering beautiful math in inline and display modes using [KaTeX](https://khan.github.io/KaTeX/) engine.
You just need to surround your math expression with `$$`, like `$$ E = mc^2 $$`.
If you leave it inside a paragraph, it will produce an inline expression, just like $$ E = mc^2 $$.

To use display mode, again surround your expression with `$$` and place it as a separate paragraph.
Here is an example:

$$
\left( \sum_{k=1}^n a_k b_k \right)^2 \leq \left( \sum_{k=1}^n a_k^2 \right) \left( \sum_{k=1}^n b_k^2 \right)
$$

Note that [KaTeX](https://khan.github.io/KaTeX/) is work in progress, so it does not support the full range of math expressions as, say, [MathJax](https://www.mathjax.org/).
Yet, it is [blazing fast](http://www.intmath.com/cg5/katex-mathjax-comparison.php).

***

## Citations

Citations are then used in the article body with the `<d-cite>` tag.
The key attribute is a reference to the id provided in the bibliography.
The key attribute can take multiple ids, separated by commas.

The citation is presented inline like this: <d-cite key="gregor2015draw"></d-cite> (a number that displays more information on hover).
If you have an appendix, a bibliography is automatically created and populated in it.

Distill chose a numerical inline citation style to improve readability of citation dense articles and because many of the benefits of longer citations are obviated by displaying more information on hover.
However, we consider it good style to mention author last names if you discuss something at length and it fits into the flow well — the authors are human and it’s nice for them to have the community associate them with their work.

***

## Footnotes

Just wrap the text you would like to show up in a footnote in a `<d-footnote>` tag.
The number of the footnote will be automatically generated.<d-footnote>This will become a hoverable footnote.</d-footnote>

***

## Code Blocks

Syntax highlighting is provided within `<d-code>` tags.
An example of inline code snippets: `<d-code language="html">let x = 10;</d-code>`.
For larger blocks of code, add a `block` attribute:

<d-code block language="javascript">
  var x = 25;
  function(x) {
    return x * x;
  }
</d-code>

***
## Structure within a Cell: real world example of graphical models with structure among the RVs

<insert graph image here>

Receptors: receives signal from cell surface
Kinase: Reads and decodes the signal
TF: Takes in the signal and triggers production of dna with dna template
Gene: DNA templates

We can incoporate such domain knowledge to impose structure on the RVs $$X_1,..,X_8$$.
A preliminary way is to partition the RV's into compartments they reside in within a cell.
Then we can model edges(pathway) that model the dependencies(communication) among the RVs(nodes).


With this structure, we can better express the joint probabilities among the RVs than with a full joint distribution table. The Factorization Law gives us a way to do so. 
The Factorization Law is a graph traversal algorithm that outputs a unique representation of the joint probability of the RVs. Concisely, we traverse the graph and identify the conditional probabilities of each node given its parent nodes, and marginal probabilities of nodes that do not have parents, then multiply all terms together for the joint probability of all nodes.

There are 3 main benefits of representing the joint distribution in this manner (with a graph structure and conditional probabilities that tie parent nodes and child nodes)

***

## 3 Main Benefits of Graphical Models

1. Save cost in representing the joint distribution
By modeling the dependencies among the RVs with a graph and conditionals, the number of parameters needed to decribe the joint distribution is much fewer than when using a full joint distribution table.

Assume all RVs are binary.

$$P(X_1) \rightarrow 1$$ parameter 
$$P(X_2) \rightarrow 1$$ parameter 
$$P(X_3|X_1) \rightarrow 2$$ parameters 
$$P(X_4|X_2) \rightarrow 2$$ parameters
$$P(X_5|X_2) \rightarrow 2$$ parameters
$$P(X_6|X_3, X_4) \rightarrow 4$$ parameters 
$$P(X_7|X_6) \rightarrow 2$$ parameters 
$$P(X_8|X_5, X_6) \rightarrow 4$$ parameters

Total = 18 parameters

Meanwhile, with a full joint distribution table, we would need $$2^8-1$$ parameters.

2. Data integration

By factoring the joint distribution into modular terms, each term becomes self-contained and we can estimate each term with only the relevant datapoints (e.g. to estimate $$P(X_8|X_5, X_6)$$ we only need data for $$X_8, X_5 and X_6$$)
Therefore, the problem of joint distribution estimcation can be modularized into smaller pieces and integrated later by multiplication.

Examples:
1) Regarding the cell molecules example, one lab can study the subtree formed by $$X_1, X_3, X_6, X_7, X_8$$ and another lab can study $$X_2, X_4, X_5$$, then fuse their estimations together by multiplying the terms by their dependencies.
2) For a cellphone user, we can separately study the distribution represented by the user's text, image and network data and fuse them toegether with a graphic model to derive the joint distribution.
3) In genomics/biology, we routinely combine various data together with graphic models.

3. Graphical models provide a generic method of representing knowledge and making inferences

We can encode our domain knowledge through priors and incorporate them into our inferece via the Bayes Theorem:
$$ p(h|d) = \frac{p(d|h)p(h)}{p(d)} $$
$$p(h|d)$$=posterior
$$p(d|h)$$=likelihood
$$p(h)$$=prior

A graphical model provides a structured and efficient way for doing these computations.
Therefore, a graphical model along with the Bayes Theorem provide a universal way of representing knowledge and computation.

***

## Layouts

The main text column is referred to as the body.
It is the assumed layout of any direct descendants of the `d-article` element.

<div class="fake-img l-body">
  <p>.l-body</p>
</div>

For images you want to display a little larger, try `.l-page`:

<div class="fake-img l-page">
  <p>.l-page</p>
</div>

All of these have an outset variant if you want to poke out from the body text a little bit.
For instance:

<div class="fake-img l-body-outset">
  <p>.l-body-outset</p>
</div>

<div class="fake-img l-page-outset">
  <p>.l-page-outset</p>
</div>

Occasionally you’ll want to use the full browser width.
For this, use `.l-screen`.
You can also inset the element a little from the edge of the browser by using the inset variant.

<div class="fake-img l-screen">
  <p>.l-screen</p>
</div>
<div class="fake-img l-screen-inset">
  <p>.l-screen-inset</p>
</div>

The final layout is for marginalia, asides, and footnotes.
It does not interrupt the normal flow of `.l-body` sized text except on mobile screen sizes.

<div class="fake-img l-gutter">
  <p>.l-gutter</p>
</div>

***

## Print

Finally, you can easily get a PDF or printed version of the notes by simply hitting `ctrl+P` (or `⌘+P` on macOS).
